system/services/routeHandlers/AssetRouteHandler.cfc

```diff
@@ -93,6 +105,10 @@ component implements="iRouteHandler" singleton=true presideService=true {
 			link = event.getSiteUrl( includePath=false, includeLanguageSlug=false ) & link;
 		}
 
+		if ( Len( tenantId ) ) {
+			link &= "?tenantId=" & tenantId;
```

Comment:
Do not assume there is not already a query string (even though there almost certainly isn't). So the delimiter would need to be an & if detecting that there is already a qs.